src/lib.rs

```diff
@@ -18,356 +17,446 @@
-                success: Ordering,
-                failure: Ordering,
-            ) -> Result<$T, $T>;
-        }
-    };
-
-    (binary: $T:ty, $s_atomic:expr) => {
-        trait_decl!(base: $T, $s_atomic);
-
-        doc_comment! {
-            concat!("Performs a bitwise \"and\" on the current value and the
-argument `val`, snd sets the new value to the result.
-
-Returns the previous value.
-
-Ordering arguments are ignored by non-atomic types.
-
-See also: [`", $s_atomic, "::fetch_and`]"),
-            fn fetch_and(&self, val: $T, order: Ordering) -> $T;
-        }
-
-        doc_comment! {
-            concat!("Performs a bitwise \"nand\" on the current value and the
-argument `val`, snd sets the new value to the result.
-
-Returns the previous value.
-
-Ordering arguments are ignored by non-atomic types.
-
-See also: [`", $s_atomic, "::fetch_nand`]"),
-            fn fetch_nand(&self, val: $T, order: Ordering) -> $T;
-        }
-
-        doc_comment! {
-            concat!("Performs a bitwise \"or\" on the current value and the
-argument `val`, snd sets the new value to the result.
-
-Returns the previous value.
-
-Ordering arguments are ignored by non-atomic types.
-
-See also: [`", $s_atomic, "::fetch_or`]"),
-            fn fetch_or(&self, val: $T, order: Ordering) -> $T;
-        }
-
-        doc_comment! {
-            concat!("Performs a bitwise \"xor\" on the current value and the
-argument `val`, snd sets the new value to the result.
-
-Returns the previous value.
-
-Ordering arguments are ignored by non-atomic types.
-
-See also: [`", $s_atomic, "::fetch_xor`]"),
-            fn fetch_xor(&self, val: $T, order: Ordering) -> $T;
-        }
-    };
-
-    (int: $T:ty, $s_atomic:expr) => {
-        trait_decl!(binary: $T, $s_atomic);
-
-        doc_comment! {
-            concat!("Increments the current value by `val`, wrapping on overflow.
-
-Returns the previous value.
-
-Ordering arguments are ignored by non-atomic types.
-
-See also: [`", $s_atomic, "::fetch_add`]"),
-            fn fetch_add(&self, val: $T, order: Ordering) -> $T;
-        }
-
-        doc_comment! {
-            concat!("Decrements the current value by `val`, wrapping on overflow.
-
-Returns the previous value.
-
-Ordering arguments are ignored by non-atomic types.
-
-See also: [`", $s_atomic, "::fetch_sub`]"),
-            fn fetch_sub(&self, val: $T, order: Ordering) -> $T;
-        }
-    };
-}
+/// Marks that a type can be viewed as a set of bits.
+///
+/// `bool` and all integer fundamentals implement this.
+///
+/// ```rust
+/// # use core::sync::atomic::*;
+/// # use radium::Radium;
+/// let num: AtomicUsize = AtomicUsize::new(0);
+/// Radium::fetch_or(&num, 2, Ordering::Relaxed);
+/// ```
+///
+/// Pointers do not. This will cause a compiler error.
+///
+/// ```rust,compile_fail
+/// # use core::sync::atomic::*;
+/// # use radium::Radium;
+/// let mut data = 0usize;
+/// let ptr: AtomicPtr<usize> = AtomicPtr::new(Default::default());
```

Comment:
I don't think `*mut usize` implements `Default`, though `AtomicPtr<usize>` does.

IIRC this will work if you remove the `AtomicPtr::new(...)` call.

```suggestion
/// let ptr: AtomicPtr<usize> = Default::default();
```